conftest.py

```diff
@@ -85,6 +85,13 @@ def pytest_configure(config):
         "network: mark a test for execution if network available."
     )
 
+    # Use matplotlib agg backend during the tests
+    try:
+        import matplotlib
+        matplotlib.use('agg')
+    except ImportError:
+        pass
```

Comment:
Can we place this in `sklearn/conftest.py`?

REF: https://github.com/scikit-learn/scikit-learn/issues/18865

Reply:
#18865 was merged in main. I tried to update this PR accordingly. Let's see if the CI is happy.